### DISCUSSION TITLE
[Feature]: Allow Subtitles (textTracks) with M3U8 Videos

### DISCUSSION THREAD

**Discussion by nicolaosm** [TheWidlarzGroup/react-native-video#4985] [Ideas]
### What problem are you trying to solve?

Many times the M3U8 server does not allow subtitles in the m3u8 file and we need to provide it separately.

### Proposed solution

Allow Subtitles (textTracks) to work with M3U8 videos.

[a comment by an automated account is not shown]

**Comment by moskalakamil** [comment]
Hi @nicolaosm, we invite you to add a comment to our [roadmap](https://github.com/TheWidlarzGroup/react-native-video/discussions/3351). If you want to have this ASAP, please [contact us](https://thewidlarzgroup.com/?utm_source=github&utm_medium=rnv#Contact) for paid development

**Comment by nicolaosm** [comment]
Thanks @moskalakamil !

**Comment by TWG-AI** [comment]
v7 supports external subtitles alongside any source (including M3U8) via `externalSubtitles` in the video config:

```ts
useVideoPlayer({
  uri: 'https://example.com/stream.m3u8',
  externalSubtitles: [
    { uri: 'https://example.com/subs_en.vtt', label: 'English', language: 'en', type: 'vtt' },
    { uri: 'https://example.com/subs_es.srt', label: 'Español', language: 'es', type: 'srt' },
  ],
})
```

Supported formats: VTT, SRT, SSA, ASS. iOS only supports VTT for HLS streams.

> I'm an AI — verify supported types and platform notes in the v7 docs.
